src/js/components/WaitForSettings/WaitForSettings.js

```diff
@@ -0,0 +1,52 @@
+import React, { Component } from 'react';
+import { isActiveZoneOnCloudflare } from '../../selectors/activeZone';
+import Loading from 'cf-component-loading';
+import Text from 'cf-component-text';
+import { FormattedMessage, injectIntl } from 'react-intl';
+
+class WaitForSettings extends Component {
+
+  render() {
+    let activeZone = this.props.activeZone;
+    let analytics = (typeof this.props.analytics != 'undefined' ? this.props.analyics : true);
+    let settings = (typeof this.props.settings != 'undefined' ? this.props.settings : true);
```

Comment:
This is not readable. The viewer should look inside the view to understand whats going on.
`<WaitForSettings activeZone={activeZone} settings={getAllZoneSettingsForZoneId(activeZone.id, zoneSettings)} pluginSettings={getPluginSettingsForZoneId(activeZone.id, pluginSettings)}>`

That is why I suggested a different blueprint such as
`<WaitForSettings waitZone=true waitSettings=true waitPluginSettings=true waitAnalytics=false/>`

I'm not a fan of implicit code. We can make `WaitForSettings` really dumb and calculate all of the fields in every page explicitly 

```
let shouldWaitAnalytics = (typeof this.props.analytics == 'undefined' ? true : false);
let shouldWaitSettings = (typeof this.props.settings == 'undefined' ? true : false);
let shouldWaitPluginSettings = (typeof this.props.pluginSettings == 'undefined' ? true : false);
```

Afterwards the blueprint will like

``<WaitForSettings waitZone={shouldWaitZone} waitSettings={shouldWaitSettings} waitPluginSettings={shouldWaitPluginSettings} waitAnalytics=false}/>`` (for non anaytic pages)